src/execution/operator/scan/physical_table_scan.cpp

```diff
@@ -120,6 +123,12 @@ SourceResultType PhysicalTableScan::GetData(ExecutionContext &context, DataChunk
 		break;
 	}
 
+	if (function.ordinality_data.ordinality_request == Ordinality_request_t::REQUESTED) {
+		idx_t ordinality = chunk.size();
+		function.ordinality_data.SetOrdinality(chunk, g_state.ordinality_idx, ordinality);
```

Comment:
For `PhysicalTableScan` - I think we can instead perform this using a window function. This will currently be incorrect when multi-threading.